core/domain/user_services.py

```diff
@@ -170,6 +172,18 @@ def validate(self):
         if self.role not in role_services.PARENT_ROLES:
             raise utils.ValidationError('Role %s does not exist.' % self.role)
 
+        if not isinstance(self.creator_dashboard_display, basestring):
+            raise utils.ValidationError(
+                'Expected dashboard display preference to be a string, '
+                'received %s' % self.creator_dashboard_display)
+        if (self.creator_dashboard_display !=
+                feconf.CREATOR_DASHBOARD_DISPLAY_PREF_CARD and
+                self.creator_dashboard_display !=
+                feconf.CREATOR_DASHBOARD_DISPLAY_PREF_LIST):
+            raise utils.ValidationError(
+                'Dashboard display preferences %s does not exist.'
```

Comment:
'%s is not a valid value for the dashboard display preferences.'

Reply:
Done